src/test/java/rocks/inspectit/gepard/agent/internal/configuration/util/ConfigurationMapperTest.java

```diff
@@ -2,29 +2,23 @@
 
 import static org.junit.jupiter.api.Assertions.*;
 
+import java.io.IOException;
+import java.util.List;
 import org.junit.jupiter.api.Test;
 import rocks.inspectit.gepard.agent.internal.configuration.exception.CouldNotDeserializeConfigurationException;
 import rocks.inspectit.gepard.agent.internal.configuration.model.InspectitConfiguration;
+import rocks.inspectit.gepard.agent.internal.configuration.model.instrumentation.InstrumentationConfiguration;
+import rocks.inspectit.gepard.agent.internal.configuration.model.instrumentation.Scope;
 
```

Comment:
We might have to talk about missing properties and unknown properties... 
What happens at the moment, if we´ve got for example the json prop "test" in the string?
What happens, if "scope" is missing? What happens, if scope is an array? What happens, if we receive just {}?
We might have to define this and write tests for it? Maybe this reveals unwanted behavior in the serialization.

Reply:
- Invalid strings, e.g. with a property "test", will throw an exception
- Arrays and lists are handled the same within JSON :)
- We should add property validation in the future, so properties like "scopes" should not contain null values. However, for this we need to add a new validation dependency. Also validation should be made by the configuration server rather than the agent
- I added one test for an empty Object {}. Currently, it's a valid object, which will result in just calling the default constructors of all properties.